src/Microsoft.ML.LightGBM/LightGbmBinaryTrainer.cs

```diff
@@ -49,17 +49,17 @@ private static VersionInfo GetVersionInfo()
                 loaderAssemblyName: typeof(LightGbmBinaryModelParameters).Assembly.FullName);
         }
 
-        protected override uint VerNumFeaturesSerialized => 0x00010002;
-        protected override uint VerDefaultValueSerialized => 0x00010004;
-        protected override uint VerCategoricalSplitSerialized => 0x00010005;
+        internal override uint VerNumFeaturesSerialized => 0x00010002;
+        internal override uint VerDefaultValueSerialized => 0x00010004;
+        internal override uint VerCategoricalSplitSerialized => 0x00010005;
         public override PredictionKind PredictionKind => PredictionKind.BinaryClassification;
 
         internal LightGbmBinaryModelParameters(IHostEnvironment env, InternalTreeEnsemble trainedEnsemble, int featureCount, string innerArgs)
             : base(env, RegistrationName, trainedEnsemble, featureCount, innerArgs)
         {
         }
 
-        private LightGbmBinaryModelParameters(IHostEnvironment env, ModelLoadContext ctx)
+        internal LightGbmBinaryModelParameters(IHostEnvironment env, ModelLoadContext ctx)
```

Comment:
>internal [](start = 8, length = 8)

This is private in all other classes...see Ranking and Regression one. Do we want it to be internal?